server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -3017,6 +3037,26 @@ public InputStream maybeRateLimitRestores(InputStream stream) {
         );
     }
 
+    public OffsetRangeInputStream maybeRateLimitRemoteUploadTransfers(OffsetRangeInputStream offsetRangeInputStream) {
+        return maybeRateLimitRemoteTransfers(offsetRangeInputStream, () -> remoteUploadRateLimiter, remoteUploadRateLimitingTimeInNanos);
+    }
+
+    public OffsetRangeInputStream maybeRateLimitRemoteDownloadTransfers(OffsetRangeInputStream offsetRangeInputStream) {
+        return maybeRateLimitRemoteTransfers(
+            maybeRateLimitRemoteTransfers(offsetRangeInputStream, () -> remoteDownloadRateLimiter, remoteDownloadRateLimitingTimeInNanos),
+            recoverySettings::rateLimiter,
+            remoteDownloadRateLimitingTimeInNanos
+        );
+    }
+
+    public InputStream maybeRateLimitRemoteDownloadTransfers(InputStream inputStream) {
+        return maybeRateLimit(
+            maybeRateLimit(inputStream, () -> remoteDownloadRateLimiter, remoteDownloadRateLimitingTimeInNanos),
+            recoverySettings::rateLimiter,
+            remoteDownloadRateLimitingTimeInNanos
+        );
```

Comment:
Why do we need to rate limit a rate limited stream again?

Reply:
https://github.com/opensearch-project/OpenSearch/pull/9448/files#r1303554896